sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidJoinRule.java

```diff
@@ -397,15 +297,26 @@ static class ConditionAnalysis
      */
     private final List<RexLiteral> literalSubConditions;
 
+    /**
+     * Sub-conditions that cannot be handled by the DruidJoinRule.
+     */
+    private final List<RexNode> unsupportedSubConditions;
```

Comment:
i can't help but wonder if there is a better name for this collection of stuff, 'unsupported' feels a bit off because we don't completely give up if there is some stuff in here... but i'm having trouble of thinking of what might be better

maybe something about 'non-native' or ... 'indirect' .. or idk naming is the worst